src/Host/Client/Impl/Session/RSessionEvaluationCommands.cs

```diff
@@ -46,31 +49,71 @@ public static class RSessionEvaluationCommands {
             return evaluation.ExecuteAsync(script);
         }
 
-        public static Task PreviousPlotAsync(this IRExpressionEvaluator evaluation) {
-            var script = "rtvs:::graphics.ide.previousplot()\n";
+        public static Task PreviousPlotAsync(this IRExpressionEvaluator evaluation, Guid deviceId) {
+            Debug.Assert(deviceId != Guid.Empty);
+            var script = Invariant($"rtvs:::graphics.ide.previousplot({deviceId.ToString().ToRStringLiteral()})\n");
             return evaluation.ExecuteAsync(script);
         }
 
-        public static Task ClearPlotHistoryAsync(this IRExpressionEvaluator evaluation) {
-            var script = "rtvs:::graphics.ide.clearplots()\n";
+        public static Task ClearPlotHistoryAsync(this IRExpressionEvaluator evaluation, Guid deviceId) {
+            Debug.Assert(deviceId != Guid.Empty);
+            var script = Invariant($"rtvs:::graphics.ide.clearplots({deviceId.ToString().ToRStringLiteral()})\n");
             return evaluation.ExecuteAsync(script);
         }
 
-        public static Task RemoveCurrentPlotAsync(this IRExpressionEvaluator evaluation) {
-            var script = "rtvs:::graphics.ide.removeplot()\n";
+        public static Task RemoveCurrentPlotAsync(this IRExpressionEvaluator evaluation, Guid deviceId, Guid plotId) {
+            Debug.Assert(deviceId != Guid.Empty);
+            Debug.Assert(plotId != Guid.Empty);
+            var script = Invariant($"rtvs:::graphics.ide.removeplot({deviceId.ToString().ToRStringLiteral()}, {plotId.ToString().ToRStringLiteral()})\n");
             return evaluation.ExecuteAsync(script);
         }
 
+        public static Task CopyPlotAsync(this IRExpressionEvaluator evaluation, Guid sourceDeviceId, Guid sourcePlotId, Guid targetDeviceId) {
+            Debug.Assert(sourceDeviceId != Guid.Empty);
+            Debug.Assert(sourcePlotId != Guid.Empty);
+            Debug.Assert(targetDeviceId != Guid.Empty);
+            var script = Invariant($"rtvs:::graphics.ide.copyplot({sourceDeviceId.ToString().ToRStringLiteral()}, {sourcePlotId.ToString().ToRStringLiteral()}, {targetDeviceId.ToString().ToRStringLiteral()})\n");
+            return evaluation.ExecuteAsync(script);
+        }
+
+        public static Task SelectPlotAsync(this IRExpressionEvaluator evaluation, Guid deviceId, Guid plotId) {
+            Debug.Assert(deviceId != Guid.Empty);
+            Debug.Assert(plotId != Guid.Empty);
+            var script = Invariant($"rtvs:::graphics.ide.selectplot({deviceId.ToString().ToRStringLiteral()}, {plotId.ToString().ToRStringLiteral()})\n");
+            return evaluation.ExecuteAsync(script);
+        }
+
+        public static async Task ActivatePlotDeviceAsync(this IRExpressionEvaluator evaluation, Guid deviceId) {
+            Debug.Assert(deviceId != Guid.Empty);
+            await evaluation.ExecuteAsync(Invariant($"rtvs:::graphics.ide.setactivedeviceid({deviceId.ToString().ToRStringLiteral()})"));
```

Comment:
`ExecuteAsync` and `EvaluateAsync` have overloads that take `FormattableString` and apply `Invariant` to it, so you don't have to do it at call site.